### PR TITLE
Auto-update signature count on petition page.

### PR DESCRIPTION
**Pull request by dracos**
This fetches the JSON of the petition every 10 seconds and updates the count, and progress bar, accordingly, with a pretty counting animation.

### CODE REVIEW

File: app/helpers/application_helper.rb [hunk 1]
```diff
@@ -23,6 +23,11 @@ def home_page?
     params.values_at(:controller, :action) == %w[pages index]
   end
 
+  def open_petition_page?
+    # XXX What should go here? Same as petition_page? plus petition.open? Is that allowed here?
+    true
```
Comment (pixeltrix) [inline app/helpers/application_helper.rb]: The method `petition_page?` is just a check for the controller and action name but we know that if it's true then the `@petition` instance variable is set so just use the `Petition#open?` method to achieve what you need, e.g:

```
petition_page? && @petition.open?
```

Hopefully that's clearer?

**Comment by pixeltrix** [comment]
The reason that `archived_petition_page` isn't in the fragment.yml file is that the head section isn't dependent on that - it's got `:url` in the list so would change anyway.

File: app/helpers/application_helper.rb (hunk 1 above)
Comment (dracos) [inline app/helpers/application_helper.rb]: Ah, great, thanks, I still don't get my head around where variables are magically available in ruby :)

**Comment by dracos** [comment]
Ah, of course, makes sense, thanks. Think only thing I don't understand now is the `Keys` class in CacheHelper, but I don't really need to to tidy this up ;) Fixes pushed.

**Comment by pixeltrix** [comment]
> Think only thing I don't understand now is the Keys class in CacheHelper

That's just a simple decorator around the template context so that we can return the values in the way that we need. Mostly it's simple delegations but some are slightly more complicated. It then creates the cache key for each item in the list of dependencies.

**Comment by pixeltrix** [comment]
@dracos one more thing - I think creating a signature count API would be useful - there's no need to pull all of the JSON data just to get the signature count.

**Comment by pixeltrix** [comment]
Closing in favour of the fixed #512 - thanks @dracos

**Pull request closed without merging**